config/core/build-configs.yaml

```diff
@@ -834,6 +834,11 @@ build_configs:
       clang-15:
         build_environment: clang-15
         architectures: *arch_clang_configs
+      rustc-1.62:
+        build_environment: rustc-1.62
+        fragments: [rust, rust-samples, kselftest]
+        architectures:
+          x86_64: *x86_64_arch
```

Comment:
Keep in mind that this anchor expands to:
```yaml
        x86_64: &x86_64_arch
          base_defconfig: 'x86_64_defconfig'
          extra_configs:
            - 'allmodconfig'
            - 'allnoconfig'
            - 'x86_64_defconfig+x86-chromebook+kselftest'
            - 'x86_64_defconfig+x86-chromebook+amdgpu'
```
which includes `allmodconfig`.  Is it really worth doing a huge allmodconfig build with Rust?  If all the Rust-related parts of the code are already turned on with the extra fragments then this should get you covered already.  Maybe it's fine to have allmodconfig builds for mainline and next for completeness, but for the subsystem branches from the `rust-for-linux` tree I think that would be overkill.  Keeping the build configs optimised helps with lower latency for the CI loop and saves some build resources.

Reply:
Some Rust abstractions may only get enabled if their subsystem is enabled. However, I would expect maintainers that start using Rust in their subsystem to add a `CONFIG_RUST=y` to test that part of their subsystem, rather than us having to add everything here. Does that make sense for KernelCI?

So I think for testing the general Rust infrastructure a `defconfig` is fine. Additionally, we are still upstreaming the basics, so we don't really have Rust-subsystems enabled yet, so I don't want to feel bad spending resources... :) I assume it is not a big deal to change the config in the future, right?

Having said that, of course, an `allmodconfig` somewhere with Rust enabled, in particular for `rust-next` (or `linux-next` as a shared Rust one for everybody), never hurts as a "final check" and to spot edge cases.

Reply:
Thanks for the extra context, yes feel free to submit PRs as and when Rust support expands in the kernel to make the KernelCI coverage grow accordingly.  And yes I think linux-next allmodconfig with Rust enabled would be quite useful at some stage, rather than the `rust-next` branch.  The advantage of keeping the builds for the `rust-for-linux` tree lightweight is that you can more easily do several pushes during the same day and get some quick turnaround, so `allmodconfig` might get in the way rather than help there.  And then `linux-next` builds happen once a day so full coverage there is very useful to everyone, especially once we start seeing patches with Rust code that don't go through the `rust-for-linux` tree (maybe that's already the case?).